[EDU-3139] feat: add waf rule set (#522)
aziontech/docs: src/content/docs/en/pages/secure-journey/advanced-configurations/create-waf-rule-set.mdx

---
title: How to create a WAF rule set
description: Create rule sets on Web Application Firewall (WAF).
meta_tags: edge, secure, waf, rule set, rules engine
namespace: docs_secure_waf_rule_set
menu_namespace: secureMenu
permalink: /documentation/products/guides/secure/create-waf-rule-set/
---

> [Web Application Firewall (WAF)](/en/documentation/products/edge-firewall/web-application-firewall/) allows you to create rule sets regarding possible [threat types](/en/documentation/products/edge-firewall/web-application-firewall/#waf-main-settings). You can set these rule sets on Rules Engine for Edge Firewall to 

Author: ```suggestion
[Web Application Firewall (WAF)](/en/documentation/products/edge-firewall/web-application-firewall/) allows you to create rule sets regarding possible [threat types](/en/documentation/products/edge-firewall/web-application-firewall/#waf-main-settings). You can set these rule sets on Rules Engine for Edge Firewall to execute their behaviors.
```

> This guide will create a rule set to block threats from the **SQL Injection** family to prevent attacks that present the `select` word, for example. After [creating an edge firewall](/en/documentation/products/guides/secure/configure-main-settings/) and [activating the WAF module](/en/documentation/products/guides/secure/configure-main-settings/#modules), follow the steps described next.

Reviewer: ```suggestion
This guide will create a rule set to block threats from the **SQL Injection** family to prevent attacks that present the `select` word. After [creating an edge firewall](/en/documentation/products/guides/secure/configure-main-settings/) and [activating the WAF module](/en/documentation/products/guides/secure/configure-main-settings/#modules), follow the steps described next.
```

> This guide will create a rule set to block threats from the **SQL Injection** family to prevent attacks that present the `select` word, for example. After [creating an edge firewall](/en/documentation/products/guides/secure/configure-main-settings/) and [activating the WAF module](/en/documentation/products/guides/secure/configure-main-settings/#modules), follow the steps described next.

Reviewer 2: ```suggestion
This guide will create a rule set to block threats from the **SQL Injection** family to prevent attacks that attempt a `select` query, for example. After [creating an edge firewall](/en/documentation/products/guides/secure/configure-main-settings/) and [activating the WAF module](/en/documentation/products/guides/secure/configure-main-settings/#modules), follow the steps described next.
```

---

## Via RTM

1. [Access RTM](/en/documentation/products/guides/how-to-access-rtm/) > **WAF Rules**.
2. Click the **Add WAF** button.
3. On the **Main Settings** tab, give your WAF rule set a unique and easy-to-remember name. In this case, `SQL Injection rule set`.

> 3. On the **Threat Type Configuration** section, look for the **SQL Injection** option.
> 4. Choose the **Sensitivity** option **High** for this threat so WAF doesn't allow requests considered as an SQL Injection threat.
> 5. Make sure the **Active** switch is enabled.
> 6. Click the **Save** button.

Reviewer 2: ```suggestion
4. On the **Threat Type Configuration** section, look for the **SQL Injection** option.
5. Choose the **Sensitivity** option **High** for this threat so WAF doesn't allow requests considered as an SQL Injection threat.
6. Make sure the **Active** switch is enabled.
7. Click the **Save** button.
```

Repeat this process with other types of threats to which you want to create a rule set.

Now, you need to apply the rule set you've created in **Rules Engine**. To do so:

1. Still on RTM, go to **Edge Firewall**.
2. Select the **Rules Engine** tab.

> 3. 4. Click the **New Rule** button.
> 5. Give your rule a name and, optionally, a description.
> 6. In the **Criteria** section, select the `Request Args` variable.
> 7. As a comparison operator, select `matches`.
> 8. As an argument, add `content-type=select`.
> 9. In the **Behaviors** section, select **Set WAF Rule Set**.
> 10. As arguments:
>     - On the first dropdown menu, select the rule set you want to use. In this case, `SQL Injection rule set`.
>     - On the second dropdown menu, select the mode you want Edge Firewall to operate on: **Learning** or **Blocking**. In this case, **Blocking**.
> 11. Click the **Save** button.

Reviewer: ```suggestion
3. Click the **New Rule** button.
4. Give your rule a name and, optionally, a description.
5. In the **Criteria** section, select the `Request Args` variable.
6. As a comparison operator, select `matches`.
7. As an argument, add `content-type=select`.
8. In the **Behaviors** section, select **Set WAF Rule Set**.
9. As arguments:
    - On the first dropdown menu, select the rule set you want to use. In this case, `SQL Injection rule set`.
    - On the second dropdown menu, select the mode you want Edge Firewall to operate on: **Learning** or **Blocking**. In this case, **Blocking**.
10. Click the **Save** button.
```

  "bypass_addresses": []
}
```

3. Copy the `id` value.
4. Run the following `POST` request to create a rule, replacing `[TOKEN VALUE]` with with your [personal token](/en/documentation/products/guides/personal-tokens/), the `<edge_firewall_id>` variable with your [edge firewall id](https://api.azion.com/#58376993-e0dd-4b51-9cd3-41d86122728f) value, and `<waf_rule_set_id>` with the id you copied:

```bash
curl --location 'https://api.azionapi.net/edge_firewall/<edge_firewall_id>/rules_engine' \
--header 'Accept: application/json; version=3' \
--header 'Authorization: Token azion5c2ecd92839e59b0cca662ada47b7382706' \
--header 'Content-Type: application/json' \
--data '{
    "name": "SQL Injection rule",
    "is_active": true,
    "behaviors": [
        {
            "name": "set_waf_ruleset",
            "argument": "<waf_rule_set_id>"
        }
    ],
    "criteria": [
        [
            {
                "variable": "request_args",
                "operator": "matches",
                "conditional": "if",
                "argument": "content-type=select"
            }
        ]
    ]
}'
```

| Key | Description |
| --- | --- |
| `name` | Name of the rule |

> | `behaviors` | Array that stores objects that define behaviors |
> | `criteria` | Array that stores objects that define criteria |

Reviewer: ```suggestion
| `behaviors` | Array that stores objects defining behaviors |
| `criteria` | Array that stores objects defining criteria |
```
